server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts

```diff
@@ -435,6 +435,7 @@ export default class ServiceProviderReferralsController {
 
     let userInputData: Record<string, unknown> | null = null
     let formError: FormValidationError | null = null
+    let serverError: FormValidationError | null = null
```

Comment:
should this be a `ServerError` type? I left it as a `FormValidationError` as it's got the same interface and because it is being displayed at the top of the form on submit, but I could duplicate that interface.

Reply:
I think even a serverError can be considered a FormValidationError if it's 4xx (generally)...

I would differentiate the errors based on if it's the user's fault or out of the user's control in which case we should tell the user to contract support (or retry).

Reply:
Yes, it'll go to a page similar to the one you showed this morning (ugly generic error page)

Reply:
ah ok, so in this case, in my opinion it still makes sense for it to be a FromValidationError.